proto/perp/v2/event.proto

```diff
@@ -50,6 +50,20 @@ message PositionChangedEvent {
 
   // The block number at which this position was changed.
   int64 block_height = 7;
+
+  // Exchanged margin is the amount of collateral received by the trader during
+  // the position change. A positve value indicates that the trader received
+  // funds, while a negative value indicates that the trader spent funds.
+  string exchanged_margin = 8 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+
+  // Change type describes the type of state transition that resulted in the
+  // position change. Change type can take the following values:
+  //
+  // ["open_position", "close_position", "liquidate", "add_margin", "remove_margin"]
+  string change_type = 9;
```

Comment:
I would use an enum here instead of a free-form string, to prevent data integrity errors downstream.